core/src/main/scala/sttp/tapir/EndpointIO.scala

```diff
@@ -79,7 +79,10 @@ object EndpointTransput {
       schema(_.modifyUnsafe[U](Schema.ModifyCollectionElements)(_.validate(v)))
 
     def description(d: String): ThisType[T] = copyWith(codec, info.description(d))
-    def default(d: T): ThisType[T] = copyWith(codec.schema(_.default(d, Some(codec.encode(d)))), info)
+    def default(d: T, encoded: Option[Any] = None): ThisType[T] = {
+      val e = Some(encoded.getOrElse(codec.encode(d)))
```

Comment:
Should we allow to pass `encoded` here? It can cause ambiguity if someone passes a value that doest not match the codec encoding

Reply:
Without this change I get following compilation error:
```
[error] /home/mariusz/sml/tapir/core/src/test/scala/sttp/tapir/annotations/DeriveEndpointIOTest.scala:244:40: unknown parameter name: encoded
[error]     val derived = EndpointInput.derived[TapirRequestTest8].asInstanceOf[EndpointInput.Query[TapirRequestTest8]]
[error]                                        ^
[error] /home/mariusz/sml/tapir/core/src/test/scala/sttp/tapir/annotations/DeriveEndpointIOTest.scala:246:43: unknown parameter name: encoded
[error]     compareTransputs(EndpointInput.derived[TapirRequestTest8], expectedInput) shouldBe true
[error]                                           ^
```

Reply:
Right, we are passing `encoded` using this method in annotations macro.

Reply:
Encoded defaults only have to be passed to schemas, not to inputs (as they have codecs). So I guess @kubinio123 is right, and we should instead opt out of using the encoded parameter on inputs (and document why :) )

Reply:
Ok, I'll revert this change.
I just need to find another way to pass the `encoded` value from the `@default` annotation to the schema without using this method.

Reply:
But the default value is already passed, being automatically encoded to the correct form?

Reply:
It turned out that the `EndpointIO.default` method was used by the macros instead of the `Schema.default`. I've updated the macros and reverted the change in `EndpointIO`.

Reply:
Reverting to using EndpointIO.default (as suggested in the review). 
Adding more tests for macro handling annotation with additional parameter.